subprojects/core/src/test/groovy/org/gradle/api/internal/AsmBackedClassGeneratorGroovyTest.groovy

```diff
@@ -187,7 +185,7 @@ class AsmBackedClassGeneratorGroovyTest extends Specification {
         i.enumProperty "foo"
 
         then:
-        thrown TypeConversionException
+        thrown IllegalArgumentException
```

Comment:
That means not so nice error messages when trying to assign the wrong enum String value in a build script. Are you sure we should do this?

Reply:
There is no difference in the actual output, it looks like this in either case:

```text
Build file '/Users/lptr/Workspace/gradle/bugs/enums-in-scripts/build.gradle' line: 30

* What went wrong:
A problem occurred evaluating root project 'enums-in-scripts'.
> Cannot convert string value 'foo' to an enum value of type 'TestEnum' (valid case insensitive values: ABC, DEF)
```

There is a difference in the stacktrace, but nothing that hinders understanding.